### PR TITLE
feat(docker): init docker-compose deps

### PR DESCRIPTION
**Pull request by mralanlee**
WIP #22

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      - MYSQL_ROOT_PASSWORD="password"
+      - MYSQL_DATABASE="bdgt"
+      - MYSQL_USER="bdgt@localhost"
+      - MYSQL_PASSWORD="password"
+    ports:
+      - "3306:3306"
+    restart: always
+
+  php:
+    image: php:7.0-apache
```
Comment (Saibamen) [inline docker-compose.yml:15]: Repo uses Laravel 8, so minimum supported PHP version is 7.3

**Review by Saibamen — CHANGES_REQUESTED** [review]

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+version: "3.4"
```
Comment (Saibamen) [inline docker-compose.yml:1]:
```suggestion
version: "3.8"
```

File: docker-compose.yml (hunk 1 above)
Comment (Saibamen) [reply docker-compose.yml:15]:
```suggestion
    image: php:7.3-apache
```

**Comment by pfaffenrodt** [comment]
I dont know, if it is already enough, by using blank php image.

use `APACHE_DOCUMENT_ROOT` env variable to point to `/app/public` and not provide two volumes ( looks wrong to me)

I suggest, create a `.docker/php/Dockerfile`.

Nice to have:
* argument to provide userid and groupid of www-data user
* npm via nvm?
* imagick
* supervisor (to boot queue worker)
* crontab
* tinker support

install all packages and NPM via nvm.

**Comment by mralanlee** [comment]
I was thinking of using another docker-compose file for node (i.e. `docker-compose.node.yml`). It doesn't seem like the project always requires hot reload or watching to occur.

> use APACHE_DOCUMENT_ROOT env variable to point to /app/public and not provide two volumes ( looks wrong to me)

Yeah, it doesn't feel ergonomic to me either. I just wanted to play with it to see how that would go, and hope to get a baseline of something working.

**Comment by Saibamen** [comment]
> * npm via nvm?

For what? Docker image will be bigger.

But if you want to test Docker images in CI with different NPM versions - it is OK for me

**Comment by pfaffenrodt** [comment]
> For what? Docker image will be bigger.

Because it is easier to use to install inside php images. nvm script is not that big. It is a suggestion.

**Comment by mralanlee** [comment]
> I dont know, if it is already enough, by using blank php image.

Maybe I could create an actual base image to pack all the required dependencies to get PHP and Laravel to work, and a base node project to tailor to the requirements a bit more. Thoughts?

**Comment by sbine** [comment]
@mralanlee I just merged #48 which was a separate first pass at this, but I would love to keep iterating on it with you–it could still be improved by switching out MySQL for MariaDB and adding SSL. No need to move forward with Apache though, I'm just going to move bdgt production over to Nginx.

**Comment by mralanlee** [comment]
Sounds good. Let me pull those changes off of that and work from that.

File: docker/php/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,24 @@
+FROM php:7.3-alpine
+
+RUN apk update && apk add --no-cache \
+      bash make mariadb-client \
+      nodejs libxml2-dev libtool \
+      npm openssl g++ \
+      git gcc imagemagick-dev \
+      libssl-dev
+
+RUN wget https://getcomposer.org/composer-stable.phar -O /usr/local/bin/composer && chmod +x /usr/local/bin/composer
```
Comment (pfaffenrodt) [inline docker/php/Dockerfile:10]:
```suggestion
COPY --from=composer:latest /usr/bin/composer /usr/bin/composer
RUN mkdir -p /var/www/.composer && chown -R www-data: /var/www/.composer && chmod -R u+w /var/www/.composer
```

**Pull request closed without merging**